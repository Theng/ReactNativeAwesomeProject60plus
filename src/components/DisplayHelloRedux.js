import React from "react";
import { Text } from "react-native";
import { useSelector } from 'react-redux';

function MyComponent() {
    const hello = useSelector(state => state.hello);
    return (
        <Text testID="hello-redux" style={{color:"red"}}>{hello.data}</Text>
    );
}

export default React.memo(MyComponent);
