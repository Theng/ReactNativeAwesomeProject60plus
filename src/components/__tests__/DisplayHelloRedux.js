import React from 'react';
import DisplayHelloRedux from '../DisplayHelloRedux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, cleanup } from 'react-native-testing-library';

afterEach(cleanup);

describe('<DisplayHelloRedux />', () => {
  // configureStore expects a list of middlewares
  const mockStore = configureStore([]);
  
  it('should match snapshot', () => {
    const store = mockStore({hello: {data:"Hello redux test"}});
    const rendered = render(
      <Provider store={store}><DisplayHelloRedux /></Provider>
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});