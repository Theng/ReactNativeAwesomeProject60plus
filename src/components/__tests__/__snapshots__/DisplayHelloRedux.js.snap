// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayHelloRedux /> should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "red",
    }
  }
  testID="hello-redux"
>
  Hello redux test
</Text>
`;
